src/server.c

```diff
@@ -6962,7 +6962,6 @@ int main(int argc, char **argv) {
             server.exec_argv[1] = zstrdup(server.configfile);
             j = 2; // Skip this arg when parsing options
         }
-        int handled_last_config_arg = 1;
```

Comment:
**Just for the record** I'm again this revert.  Passing `redis-server --save --loglevel vebose` seems like a bug to me. This "implicit empty arg" feature is something I suggest removing even if it's a breaking change.

Reply:
In this context it's even worse, because `--save` implies "don't save". Skipping the value is wrong and goes against the documentation as well, so it should be fixed. A bit late, but maybe 7.0.1 is still reasonable timing.

Reply:
so @yossigo we agree that this is a bug and we should fix it (introducing a breaking change), and you argue that 7.0.1 is kinda still an early version of 7.0 and we could still afford to introduce such a fix?

the alternative is to merge just the fix in config.c now, and postpone the fix in server.c for 7.2.

note that in some cases, it could be an "innocent" mistake, not a plain bad command line.
for instance, someone composing and passing a single command line string to a python function that uses subprocess.popen, he'll then use `.split(" ")`.
so the code that composes the command line string takes a "save" config array, concatenates it to the command line string, and then if it was empty, the splitting could result in that case.

Reply:
@oranagra Yes, I think we could still fix that.
BTW In the example you brought, it's even more likely that the empty string will be retained (unless `shell=True` was used).

Reply:
@yossigo i'm not sure what `shell=True` has to do with it.
i was thinking of something like
```
save = []
s = "--save"
for ss in save:
    s += " %s" %s
s += " --loglevel verbose"
```
then passing `s` to some method that does:
```
cmd = "redis-server " + s
subprocess.popen(cmd.split(" "))
```

Reply:
If someone wrote the bug above, I don't think we should have explicit code to detect it, especially if it causes other issues this PR tries to fix.
As a side note maybe it's worth deprecating the empty sting option in `save` and having a special save string `save off` meaning no save. Then the code above should be
```python
save = ['off']
```
which will avoid the bug.

Reply:
I discussed this again with Yossi, we wanna proceed fixing this bug.
@enjoy-binbin please undo the revert of your last commit.

Reply:
ok, i revert the commit